husarion_ugv_lights/include/husarion_ugv_lights/led_components/segment_layer_interface.hpp

```diff
@@ -0,0 +1,134 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HUSARION_UGV_LIGHTS_LED_COMPONENTS_SEGMENT_LAYER_INTERFACE_HPP_
+#define HUSARION_UGV_LIGHTS_LED_COMPONENTS_SEGMENT_LAYER_INTERFACE_HPP_
+
+#include <cstdint>
+#include <rclcpp/time.hpp>
+#include <vector>
+
+#include "yaml-cpp/yaml.h"
+
+#include "pluginlib/class_loader.hpp"
+
+#include "husarion_ugv_lights/animation/animation.hpp"
+
+namespace husarion_ugv_lights
+{
+
+/**
+ * @brief Interface for LED segment layers
+ */
+class SegmentLayerInterface
+{
+public:
+  /**
+   * @brief Parses basic parameters of the LED segment
+   *
+   * @param num_led number of LEDs
+   * @param invert_led_order if true will invert the order of LEDs
+   * @param controller_frequency frequency at which animation will be updated.
+   *
+   * @exception std::runtime_error or std::invalid_argument if missing required description key or
+   * key couldn't be parsed
+   */
+  SegmentLayerInterface(
+    const std::size_t num_led, const bool invert_led_order, const float controller_frequency)
+  : controller_frequency_(controller_frequency),
+    invert_led_order_(invert_led_order),
+    num_led_(num_led)
+  {
+    animation_loader_ = std::make_shared<pluginlib::ClassLoader<husarion_ugv_lights::Animation>>(
+      "husarion_ugv_lights", "husarion_ugv_lights::Animation");
+  }
+
+  virtual ~SegmentLayerInterface()
+  {
+    // make sure that animations are destroyed before pluginlib loader
+    animation_.reset();
+    animation_loader_.reset();
+  }
+
+  /**
+   * @brief Overwrite current animation
+   *
+   * @param animation_description YAML description of the animation. Must contain 'type' key -
+   * pluginlib animation type
+   * @param repeating if true, will set the default animation for the panel
+   *
+   * @exception std::runtime_error if 'type' key is missing, given pluginlib fails to load or
+   * animation fails to initialize
+   */
+  virtual void SetAnimation(
+    const std::string & type, const YAML::Node & animation_description,
+    [[maybe_unused]] const bool repeating, const std::string & param = "") = 0;
```

Comment:
```suggestion
  virtual void SetAnimation(
    const std::string & type, const YAML::Node & animation_description,
    const bool /* repeating */, const std::string & param = "") = 0;
```
We use this syntax instead